modules/core/05-port/types/module.go

```diff
@@ -145,3 +145,8 @@ type PacketDataUnmarshaler interface {
 	// the packet data can be unmarshaled based on the channel version.
 	UnmarshalPacketData(ctx sdk.Context, portID string, channelID string, bz []byte) (any, string, error)
 }
+
+type PacketUnmarshalarModule interface {
```

Comment:
Why was this needed?

Reply:
Ok, I understand. That makes sense. The question then becomes: do we actually need (or want) to do this differently than for the transfer module, which ensures this with simply
```go
	_ porttypes.IBCModule             = (*IBCModule)(nil)
	_ porttypes.PacketDataUnmarshaler = (*IBCModule)(nil)
```

It would also work to just do the same for the IBCMiddleware struct in PFM, no?